app/js/components/LabResultsList.jsx

```diff
@@ -327,7 +327,12 @@ export class LabResultsList extends PureComponent {
     const getPatientLabResults = () => {
       // build a list of all obs from the encounters that are of type LabSet or Test
       return encounters.reduce((acc, encounter) => {
-        const obs = encounter.obs ? encounter.obs.filter(o => isLabSet(o) || isTest(o)) : [];
+        let obs = encounter.obs ? encounter.obs : [];
+        // flatten obs groups into a single list of obs
+        while (obs.some(o => o.groupMembers)) {
+          obs = obs.flatMap(o => (o.groupMembers ? o.groupMembers : o));
+        }
+        obs = obs.filter(o => isLabSet(o) || isTest(o));
```

Comment:
@cioan @brandones  here's my change... it's not recursive nor functional but I find this easier to follow...

Reply:
Very nice :ok_hand: